cert/spec.go

```diff
@@ -138,9 +141,15 @@ func newSpecFromPath(path string) (*Spec, error) {
 		return nil, err
 	}
 
+	specStat, err := os.Stat(path)
+	if err != nil {
+		// Hit the race; we read the file but someone wiped it.
+		return nil, err
```

Comment:
I realize this probably applies to many other places in the repo, but its nice to add context to errors (very useful in debugging), for example: `errors.Wrap("newSpecFromPath: failed to stat cert spec", err)`

Reply:
I'm viewing this as equivalent to python's `raise from` for exception chaining.  Correct me if I'm wrong please.

I'm a bit wary of attaching a stack trace to it, but the exception 'bubbling' up that occurs is already wrapping.

If the error dumping doesn't include tracebacks by default, I'm interested- if it does, I'm interested in ways to turn that off.

Reply:
Go doesn't attach context to errors the way Python does to exceptions. Here's an example: https://play.golang.org/p/8HZVm8-egIu

There aren't any tracebacks by default, though a `panic()` will show some traceback information. When an error is logged but doesn't panic, there is no traceback. So, for example, an error like "file not found" doesn't give the operator any clues as to what file doesn't exist under what context. If that error says "myFuncName: file %s not found", yields more clues for debugging it. To make adding this context less painful, [errors.Wrap()](https://godoc.org/github.com/pkg/errors#Wrap) (and `Wrapf()`) take a context string and an error and concat them together.

This article is a bit old, but look at the "Annotating Errors" section on https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully; it explains this better than I am :)

Either way, not a blocker for this PR, as returning naked errors (and errors with insufficient context) is all over this codebase and they should probably all be cleaned up.